docs/investigations/graphql-partial-updates/graphql-partial-updates.md

```diff
@@ -0,0 +1,156 @@
+# GraphQL Partial Updates
+
+## Overview
+
+This documents describes and compares different approaches to designing partial updates in graphql schema. Most of presented approaches have a link to an example they were based on. 
+
+## Partial update mutation designs
+
+### 1. Simple Patch
+
+#### Schema
+```graphql
+input ApplicationPatchInput {
+    name: String
+    description: String
+    webhooks: [WebhookInput!]
+    healthCheckURL: String
+    apis: [APIDefinitionInput!]
+    eventAPIs: [EventAPIDefinitionInput!]
+    documents: [DocumentInput!]
+}
+
+type Mutation {
+    updateApplication(id: ID!, patch: ApplicationPatchInput!): Application!
+}
+```
+
+#### Description
```

Comment:
1. please add examples:
- how to set description to empty value
- how to remove all eventAPIs for Application

2. Let assume that name is required, and the description is optional. Previously such validation was done on the schema label, now we have to implement it on our own. Where this validation will be added?

3. Let assume that user wants to update the only name, can you describe how it will be handled on the resolver/service/repository level? Are we going to create a SQL query builder to update only one field for Application?

Reply:
Ad 2. In this approach we're unable to set field to `nil`, so the fields that were required during creation won't be removed during update. We're still able to set name to `""` but it's the same in current version of our API.

Ad 3. I added appendix to the doc about this.